src/decoder/mp3.rs

```diff
@@ -18,9 +18,12 @@ impl<R> Mp3Decoder<R>
 where
     R: Read + Seek,
 {
-    pub fn new(data: R) -> Result<Self, ()> {
+    pub fn new(mut data: R) -> Result<Self, R> {
+        if !is_mp3(data.by_ref()) {
+            return Err(data);
+        }
         let mut decoder = Decoder::new(data);
-        let current_frame = decoder.next_frame().map_err(|_| ())?;
+        let current_frame = decoder.next_frame().unwrap();
```

Comment:
Why do you put an unwrap here, especially when before there used to be a map_err.

Reply:
The `is_mp3` function already runs through this logic before it gets to this point so this should never return an error now. This mirrors the functionality used by the other codecs. I had to add this because the mp3 decoder used to be last but now the Symphonia one is last so this needs to return the input like the other codecs do.